[2025-07-07T17:57:19.533+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-07-07T17:57:19.562+0000] {taskinstance.py:2631} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: process_sword_messages.process_sword_messages manual__2025-07-07T17:51:53.838169+00:00 [queued]>
[2025-07-07T17:57:19.571+0000] {taskinstance.py:2631} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: process_sword_messages.process_sword_messages manual__2025-07-07T17:51:53.838169+00:00 [queued]>
[2025-07-07T17:57:19.571+0000] {taskinstance.py:2884} INFO - Starting attempt 2 of 2
[2025-07-07T17:57:19.621+0000] {taskinstance.py:2907} INFO - Executing <Task(SparkSubmitOperator): process_sword_messages> on 2025-07-07 17:51:53.838169+00:00
[2025-07-07T17:57:19.627+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=391) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-07T17:57:19.629+0000] {standard_task_runner.py:72} INFO - Started process 392 to run task
[2025-07-07T17:57:19.628+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'process_sword_messages', 'process_sword_messages', 'manual__2025-07-07T17:51:53.838169+00:00', '--job-id', '105', '--raw', '--subdir', 'DAGS_FOLDER/process_sword_messages.py', '--cfg-path', '/tmp/tmpml9mer4g']
[2025-07-07T17:57:19.630+0000] {standard_task_runner.py:105} INFO - Job 105: Subtask process_sword_messages
[2025-07-07T17:57:19.734+0000] {task_command.py:467} INFO - Running <TaskInstance: process_sword_messages.process_sword_messages manual__2025-07-07T17:51:53.838169+00:00 [running]> on host 33b1069efe97
[2025-07-07T17:57:19.879+0000] {taskinstance.py:3157} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='process_sword_messages' AIRFLOW_CTX_TASK_ID='process_sword_messages' AIRFLOW_CTX_EXECUTION_DATE='2025-07-07T17:51:53.838169+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-07T17:51:53.838169+00:00'
[2025-07-07T17:57:19.880+0000] {taskinstance.py:740} INFO - ::endgroup::
[2025-07-07T17:57:19.918+0000] {base.py:84} INFO - Retrieving connection 'spark-ebsim'
[2025-07-07T17:57:19.920+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --name arrow-spark --deploy-mode client /opt/bitnami/spark/jobs/sword_processor_job.py --input-file /opt/bitnami/spark/jobs/test_sword_messages.bin
[2025-07-07T17:57:23.531+0000] {spark_submit.py:644} INFO - Traceback (most recent call last):
[2025-07-07T17:57:23.532+0000] {spark_submit.py:644} INFO - File "/opt/bitnami/spark/jobs/sword_processor_job.py", line 10, in <module>
[2025-07-07T17:57:23.535+0000] {spark_submit.py:644} INFO - from simulation_client_pb2 import SimToClient
[2025-07-07T17:57:23.536+0000] {spark_submit.py:644} INFO - File "/opt/bitnami/spark/jobs/simulation_client_pb2.py", line 12, in <module>
[2025-07-07T17:57:23.538+0000] {spark_submit.py:644} INFO - _runtime_version.ValidateProtobufRuntimeVersion(
[2025-07-07T17:57:23.539+0000] {spark_submit.py:644} INFO - File "/home/airflow/.local/lib/python3.12/site-packages/google/protobuf/runtime_version.py", line 106, in ValidateProtobufRuntimeVersion
[2025-07-07T17:57:23.540+0000] {spark_submit.py:644} INFO - _ReportVersionError(
[2025-07-07T17:57:23.540+0000] {spark_submit.py:644} INFO - File "/home/airflow/.local/lib/python3.12/site-packages/google/protobuf/runtime_version.py", line 50, in _ReportVersionError
[2025-07-07T17:57:23.540+0000] {spark_submit.py:644} INFO - raise VersionError(msg)
[2025-07-07T17:57:23.541+0000] {spark_submit.py:644} INFO - google.protobuf.runtime_version.VersionError: Detected mismatched Protobuf Gencode/Runtime major versions when loading simulation_client.proto: gencode 6.31.1 runtime 5.29.4. Same major version is required. See Protobuf version guarantees at https://protobuf.dev/support/cross-version-runtime-guarantee.
[2025-07-07T17:57:23.591+0000] {spark_submit.py:644} INFO - 25/07/07 17:57:23 INFO ShutdownHookManager: Shutdown hook called
[2025-07-07T17:57:23.593+0000] {spark_submit.py:644} INFO - 25/07/07 17:57:23 INFO ShutdownHookManager: Deleting directory /tmp/spark-ee11059e-700d-440d-bb38-cf8a18034f72
[2025-07-07T17:57:23.639+0000] {taskinstance.py:3336} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 776, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 742, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 197, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 566, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark --deploy-mode client /opt/bitnami/spark/jobs/sword_processor_job.py --input-file /opt/bitnami/spark/jobs/test_sword_messages.bin. Error code is: 1.
[2025-07-07T17:57:23.676+0000] {taskinstance.py:1242} INFO - Marking task as FAILED. dag_id=process_sword_messages, task_id=process_sword_messages, run_id=manual__2025-07-07T17:51:53.838169+00:00, execution_date=20250707T175153, start_date=20250707T175719, end_date=20250707T175723
[2025-07-07T17:57:23.779+0000] {taskinstance.py:349} INFO - ::group::Post task execution logs
[2025-07-07T17:57:23.780+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 105 for task process_sword_messages (Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark --deploy-mode client /opt/bitnami/spark/jobs/sword_processor_job.py --input-file /opt/bitnami/spark/jobs/test_sword_messages.bin. Error code is: 1.; 392)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3029, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 282, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3184, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3208, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 776, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 742, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 197, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 566, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark --deploy-mode client /opt/bitnami/spark/jobs/sword_processor_job.py --input-file /opt/bitnami/spark/jobs/test_sword_messages.bin. Error code is: 1.
[2025-07-07T17:57:23.790+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-07-07T17:57:23.814+0000] {taskinstance.py:3924} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-07T17:57:23.854+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
